base/intfuncs.jl

```diff
@@ -471,6 +469,27 @@ function ndigits0zpb(x::Base.BitUnsigned, b::Int)
     return d
 end
 
+function ndigits0zpb(x::Integer, b::Integer)
+    # precondition: b > 1
+    x == 0 && return 0
+    b = Int(b)
+
```

Comment:
I think this algorithm works only for `x` positive, in which case you should add `x = abs(x)`.